src/fileops.rs

```diff
@@ -1,27 +1,54 @@
 //! Contains operations related to the /dev/ipath character files.
-//! fcntl is not included because rust does not allow for the C-style
-//! argument list (...) outside of ffi functions. fcntl is only used once
-//! in the C version.
 
 extern crate libc;
 
 use std::ffi::CString;
 use libc::c_int;
+use std::os::unix::io::{RawFd, AsRawFd};
+use std::ops::Drop;
 
-pub fn open<T: Into<String>>(path: T, mode: c_int) -> c_int {
-  unsafe { libc::open(CString::new(path.into())
-                      .unwrap_or(CString::new("").unwrap()).as_ptr(), mode) }
+pub struct Fd(RawFd);
+
+impl Fd {
+  pub fn open<T: Into<String>>(path: T, mode: c_int) -> Option<Fd> {
+    let fd = unsafe { libc::open(CString::new(path.into())
+                .unwrap_or(CString::new("").unwrap()).as_ptr(), mode) };
+    match fd {
+      -1 => None,
```

Comment:
Yeah it makes sense, black-box. I either get a reasonable Fd, or nothing/error. Its just that, after you've gotten this Fd, and forgotten where it came from, you may also forget that it *cant* contain a -1, so you do pointless checks. The SafeFd or whatever would codify the fact that this Fd cannot contain anything invalid.

Minor issue: You could still instantiate the SafeFd manually via SafeFd { ... }

Either way is OK, just a minor bikesheddy comment.